nitrates/lib/calc_BAT_ul.py

```diff
@@ -0,0 +1,286 @@
+import numpy as np
+from astropy.table import Table, vstack
+import matplotlib.pyplot as plt
+import os
+
+#cwd = os.getcwd()
+
+#drm_dir_old = cwd+'/rsp_maskweight/'
+#drm_dir_nitrates = cwd+'/rsp_NITRATES/'
+drm_dir_old = '/gpfs/group/jak51/default/responses/rsp_maskweight/'
+drm_dir_nitrates = '/gpfs/group/jak51/default/responses/rsp_NITRATES/'
+rsps4limits ='/gpfs/group/jak51/default/rsps4limits/'
+
+# comptomized and band spectra functions
+# E is photon energy, A is normalization, E0 is pivot energy
+
+def band_Elow_spec(E, A, alpha, beta, Epeak, E0):
+    # A*((E/E0)**alpha)*exp[-(alpha+2)*E/Epeak]
+    return A*((E/E0)**(alpha))*np.exp(-(alpha+2.0)*E/Epeak)
+
+def band_Ehi_spec(E, A, alpha, beta, Epeak, E0):
+    # A*((E/E0)**beta)*exp[beta-alpha]*((alpha-beta)*(Epeak/E0)/(alpha+2))**(alpha-beta)
+    return A*((E/E0)**(beta))*np.exp(beta-alpha)*\
+            ((alpha-beta)*(Epeak/E0)/(alpha+2.0))**(alpha-beta)
+
+def band_spec(E, A, alpha, beta, Epeak, E0=100.0):
+    '''
+    Band spectra
+    '''
+
+    Ebreak = (alpha - beta)*Epeak/(alpha+2.0)
+    if np.isscalar(E):
+        if E < Ebreak:
+            return band_Elow_spec(E, A, alpha, beta, Epeak, E0)
+        else:
+            return band_Ehi_spec(E, A, alpha, beta, Epeak, E0)
+    low_inds = np.where((E<Ebreak))
+    hi_inds = np.where((E>=Ebreak))
+    spec = np.zeros_like(E)
+    spec[low_inds] = band_Elow_spec(E[low_inds], A, alpha, beta, Epeak, E0)
+    spec[hi_inds] = band_Ehi_spec(E[hi_inds], A, alpha, beta, Epeak, E0)
+    return spec
+
+
+def comp_spec(E, A, alpha, Epeak, E0=100.0):
+    '''
+    Comptomized spectra
+    '''
+
+    spec = A*((E/E0)**(alpha))*np.exp(-E*(2.+alpha)/Epeak)
+    return spec
+
+
+def get_comp_photon_fluxes(Elos, Ehis, A, alpha, Epeak, E0=100.0, Nsub_bins=10):
+    '''
+    Gets the photon flux (photons per cm2 per s) in a set of energy bins
+    for a Comptomized spectra
+    '''
+
+
+    DeltaEs = Ehis - Elos
+    NEs = len(Elos)
+
+    fluxes = np.zeros(NEs)
+
+    # loops over photon energy bins
+    for i in range(NEs):
+
+        Es = np.linspace(Elos[i], Ehis[i], Nsub_bins)
+        dE = Es[1] - Es[0]
+        # integrate spec*dE to get photon flux in bin
+        fluxes[i] = np.sum(comp_spec(Es, A, alpha, Epeak, E0=E0))*dE
+
+    return fluxes
+
+def get_band_photon_fluxes(Elos, Ehis, A, alpha, beta, Epeak, E0=100.0, Nsub_bins=10):
```

Comment:
arent a lot of these functions already defined in flux_models or something?

Reply:
I do see the band and cutoff pl models described in flux_models too. Not sure if the other functions that are defined in this file are elsewhere. Leaving this as is for now. Maybe Jimmy can help see if some of the functions are redundant.

Reply:
Yeah you could use flux models to replace a few of these functions. I don't know if we need to do that. These functions are actually commented haha